From 06af8af17587ef5ac5399211700dd3d8297968a2 Mon Sep 17 00:00:00 2001
From: ninez <johnstonljordan@gmail.com>
Date: Sat, 29 Jan 2022 09:55:56 -0500
Subject: [PATCH 209/228] pinctrl-amd.c IRQF_NO_THREAD

 - amd_pinctrl pukes on boot with threadirqs boot parameter, as seen in the
   kernel ring buffer.

[    1.926523] ------------[ cut here ]------------
[    1.926526] WARNING: CPU: 6 PID: 382 at kernel/irq/irqdesc.c:679 generic_handle_domain_irq+0x56/0x60
[    1.926536] Modules linked in: i2c_hid_acpi video(+) rng_core i2c_hid rfkill pinctrl_amd mac_hid soc_button_array acpi_cpufreq pkcs8_key_parser ipmi_devintf ipmi_msghandler crypto_user fuse bpf_preload ip_tables x_tables ext4 crc32c_generic crc16 mbcache jbd2 serio_raw atkbd libps2 sdhci_pci cqhci crc32c_intel sdhci mmc_core i8042 serio i2c_piix4 xhci_pci xhci_pci_renesas amdgpu drm_ttm_helper ttm gpu_sched
[    1.926570] CPU: 6 PID: 382 Comm: irq/7-pinctrl_a Not tainted 5.16.1-10-cachy-nspa #1 0044cfafb03d1a861fb745b57bd4b292cf1bcabe
[    1.926575] Hardware name: LENOVO 81SS/LNVNB161216, BIOS AGCN23WW(V1.06) 08/26/2019
[    1.926578] RIP: 0010:generic_handle_domain_irq+0x56/0x60
[    1.926585] Code: 00 00 0f 00 75 15 48 8b 47 78 8b 00 a9 00 00 00 10 74 08 0f 0b b8 ff ff ff ff c3 48 8b 87 a8 00 00 00 0f ae e8 ff d0 31 c0 c3 <0f> 0b eb b9 b8 ea ff ff ff c3 0f 1f 44 00 00 53 9c 58 0f 1f 44 00
[    1.926588] RSP: 0018:ffffb1c942023e18 EFLAGS: 00010046
[    1.926591] RAX: 0000000080000200 RBX: ffffb1c9404cd660 RCX: 000000000000005a
[    1.926594] RDX: 0000000000000000 RSI: 000000000000005a RDI: ffff9a14cab35000
[    1.926597] RBP: 0000000000400000 R08: 0000000010041b00 R09: ffffb1c9404cd668
[    1.926599] R10: 0000000000000000 R11: 0000000000000006 R12: 0000000000000000
[    1.926602] R13: ffff9a14c10db028 R14: 0000000000000058 R15: 0000000000000007
[    1.926604] FS:  0000000000000000(0000) GS:ffff9a187cb80000(0000) knlGS:0000000000000000
[    1.926607] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    1.926609] CR2: 00007f486bdc2a88 CR3: 00000001099be000 CR4: 00000000003506e0
[    1.926612] Call Trace:
[    1.926615]  <TASK>
[    1.926617]  amd_gpio_irq_handler+0x18b/0x410 [pinctrl_amd a5ffb4f74be4f28d68ff73fff74e255bff9bd2fb]
[    1.926625]  irq_forced_thread_fn+0x31/0x90
[    1.926629]  irq_thread+0x183/0x250
[    1.926631]  ? irq_finalize_oneshot.part.0+0xf0/0xf0
[    1.926634]  ? irq_thread_fn+0x60/0x60
[    1.926637]  ? free_percpu_irq+0x80/0x80
[    1.926639]  kthread+0x201/0x230
[    1.926643]  ? kthread_create_worker_on_cpu+0x1b0/0x1b0
[    1.926646]  ret_from_fork+0x22/0x30
[    1.926652]  </TASK>
[    1.926653] ---[ end trace 0000000000000000 ]---

Signed-off-by: ninez <johnstonljordan@gmail.com>
Signed-off-by: Jordan Johnston <johnstonljordan@gmail.com>
---
 drivers/pinctrl/pinctrl-amd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pinctrl/pinctrl-amd.c b/drivers/pinctrl/pinctrl-amd.c
index 1a7d686494ff..8024f2210643 100644
--- a/drivers/pinctrl/pinctrl-amd.c
+++ b/drivers/pinctrl/pinctrl-amd.c
@@ -1046,7 +1046,7 @@ static int amd_gpio_probe(struct platform_device *pdev)
 	}
 
 	ret = devm_request_irq(&pdev->dev, gpio_dev->irq, amd_gpio_irq_handler,
-			       IRQF_SHARED, KBUILD_MODNAME, gpio_dev);
+			       IRQF_SHARED | IRQF_NO_THREAD, KBUILD_MODNAME, gpio_dev);
 	if (ret)
 		goto out2;
 
-- 
2.36.1

